api/src/main/java/org/jenkinsci/plugins/workflow/FilePathUtils.java

```diff
@@ -39,64 +38,82 @@
 import java.util.WeakHashMap;
 import java.util.logging.Level;
 import java.util.logging.Logger;
-
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
 import jenkins.model.Jenkins;
 import org.kohsuke.accmod.Restricted;
 import org.kohsuke.accmod.restrictions.NoExternalUse;
 
-import javax.annotation.Nonnull;
-
 /**
- * @author Kohsuke Kawaguchi
+ * Candidates for inclusion in {@link FilePath}.
+ * TODO JENKINS-26096
  */
-public class FilePathPickle extends Pickle {
-
-    private static final Logger LOGGER = Logger.getLogger(FilePathPickle.class.getName());
-
-    private final String slave;
-    private final String path;
-
-    private FilePathPickle(FilePath v) {
-        // TODO JENKINS-26096 switch to FilePath.toComputer once it uses something like Listener here and so is actually reliable
-        slave = Listener.getChannelName(v.getChannel());
-        if (slave == null) {
-            throw new IllegalStateException("no known slave for " + v);
-        }
-        path = v.getRemote();
+public class FilePathUtils {
```

Comment:
The changes are needed so that code can be shared across several plugins. Formerly it was mostly in `workflow-support` (with a little copied into `workflow-api`); now there is one class defined in `workflow-api` and used in `workflow-api`, `workflow-support`, and `workflow-durable-task-step`.